File: app/src/main/java/com/kota/Bahamut/Pages/Model/MailBoxPageItem.java
package com.kota.Bahamut.Pages.Model;

import com.kota.Bahamut.ListPage.TelnetListPageItem;
import java.util.Stack;

public class MailBoxPageItem extends TelnetListPageItem {
    private static int _count = 0;
    private static Stack<MailBoxPageItem> _pool = new Stack<>();
    public String Author = null;
    public String Date = null;
    public int Size = 0;
    public String Title = null;
    public boolean isMarked = false;
    public boolean isOrigin = false;
    public boolean isRead = false;
    public boolean isReply = false;

    private MailBoxPageItem() {
    }

    public static void release() {
        synchronized (_pool) {
            _pool.clear();
        }
    }

    public static void recycle(MailBoxPageItem item) {
        synchronized (_pool) {
            _pool.push(item);
        }
    }

    public static MailBoxPageItem create() {
        MailBoxPageItem item = null;
        synchronized (_pool) {
            if (_pool.size() > 0) {
                item = _pool.pop();
            }
        }
        if (item == null) {
            return new MailBoxPageItem();
        }
        return item;
    }

    public void clear() {
        super.clear();
        this.Size = 0;
        this.isRead = false;
        this.isReply = false;
        this.Date = null;
        this.Author = null;
        this.Title = null;
    }

    /* access modifiers changed from: protected */
    public void finalize() throws Throwable {
        super.finalize();
    }

    public void set(TelnetListPageItem item) {
        super.set(item);
        if (item != null) {
            MailBoxPageItem mail_item = (MailBoxPageItem) item;
            this.Size = mail_item.Size;
            this.Date = mail_item.Date;
            this.Author = mail_item.Author;
            this.isRead = mail_item.isRead;
            this.isReply = mail_item.isReply;
            this.isMarked = mail_item.isMarked;
            this.Title = mail_item.Title;
        }
    }
}

File: app/src/main/java/com/kota/Bahamut/pages/model/MailBoxPageItem.kt
package com.kota.Bahamut.pages.model

import com.kota.Bahamut.listPage.TelnetListPageItem
import java.util.Stack

class MailBoxPageItem private constructor() : TelnetListPageItem() {
    @JvmField
    var author: String? = null
    @JvmField
    var date: String? = null
    var size: Int = 0
    @JvmField
    var title: String? = null
    @JvmField
    var isMarked: Boolean = false
    var isOrigin: Boolean = false
    @JvmField
    var isRead: Boolean = false
    @JvmField
    var isReply: Boolean = false

    override fun clear() {
        super.clear()
        this.size = 0
        this.isRead = false
        this.isReply = false
        this.date = null
        this.author = null
        this.title = null
    }

    override fun set(aData: TelnetListPageItem?) {
        super.set(aData)
        if (aData != null) {
            val mailItem = aData as MailBoxPageItem
            this.size = mailItem.size
            this.date = mailItem.date
            this.author = mailItem.author
            this.isRead = mailItem.isRead
            this.isReply = mailItem.isReply
            this.isMarked = mailItem.isMarked
            this.title = mailItem.title
        }
    }

    companion object {
        private const val COUNT = 0
        private val _pool = Stack<MailBoxPageItem?>()
        @JvmStatic
        fun release() {
            synchronized(_pool) {
                _pool.clear()
            }
        }

        @JvmStatic
        fun recycle(item: MailBoxPageItem?) {
            synchronized(_pool) {
                _pool.push(item)
            }
        }

        fun create(): MailBoxPageItem {
            var item: MailBoxPageItem? = null
            synchronized(_pool) {
                if (_pool.isNotEmpty()) {
                    item = _pool.pop()
                }
            }
            if (item == null) {
                return MailBoxPageItem()
            }
            return item
        }
    }
}
